### PR TITLE
mod_auth_mellon: add 0.14.2

### PR DESCRIPTION
**Pull request by grueni**
Needs xmlsec and lasso.
The project has no installer and so sample-manifest contains no project specific files.

### CODE REVIEW

File: components/web/apache2-modules/mod_auth_mellon/Makefile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2011, 2015, Oracle and/or its affiliates. All rights reserved.
+#
+
+include ../../../../make-rules/shared-macros.mk
+
+BITS=64
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/Makefile]:
```suggestion
BUILD_BITS=64
```
Before `include ../../../../make-rules/shared-macros.mk`?

File: components/web/apache2-modules/mod_auth_mellon/mod_auth_mellon.p5m [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# there is no install section in the MAkefile
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/mod_auth_mellon.p5m]:
```suggestion
# there is no install section in the Makefile
```

File: components/web/apache2-modules/mod_auth_mellon/mod_auth_mellon.p5m [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# we use the files from the build folders
+file files/sample-mellon.conf path=$(ETCDIR)/apache2/2.4/samples-conf.d/mellon.conf
+file build/$(MACH64)/mellon_create_metadata.sh \
+    path=$(USRDIR)/apache2/2.4/bin/mellon_create_metadata.sh mode=0755 \
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/mod_auth_mellon.p5m]: Does it need to be editable for the owner?

File: components/web/apache2-modules/mod_auth_mellon/mod_auth_mellon.p5m [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    pkg.linted.userland.action001.0=true
+file build/$(MACH64)/.libs/mod_auth_mellon.so \
+    path=$(USRDIR)/apache2/2.4/libexec/mod_auth_mellon.so \
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/mod_auth_mellon.p5m:34]: Everyone seems to be doing it like this...

File: components/web/apache2-modules/mod_auth_mellon/patches/000-remove-Wall.patch [hunk 5]
```diff
@@ -0,0 +1,16 @@
+Solaris specific patch not for upstream. Oracle Developer Studio's CC does not
+like GCC -Wall option. Remove it from Makefile.in which hardcodes it. The patch
+can be removed when Apache build switches to GCC.
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/patches/000-remove-Wall.patch]: We build with GCC, so we don't need it?

File: components/web/apache2-modules/mod_auth_mellon/patches/001-use_gnu_bin_path.patch [hunk 6]
```diff
@@ -0,0 +1,24 @@
+Solaris grep supports limited regular expressions.
+Use GNU grep in mellon_create_metadata.sh script.
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/patches/001-use_gnu_bin_path.patch:2]: Please add runtime dependency on GNU grep like this: https://github.com/OpenIndiana/oi-userland/blob/10f5892f9ab4d7cb51b52631b9f5b046b5400a97/components/archiver/zstd/zstd.p5m#L24
I think it won't be found automatically.

File: components/web/apache2-modules/mod_auth_mellon/Makefile [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+COMPONENT_PRE_CONFIGURE_ACTION = \
+	($(CLONEY) $(SOURCE_DIR) $(@D)) && \
+	(cd $(SOURCE_DIR); ./autogen.sh)
+
+CONFIGURE_OPTIONS.APXS= --with-apxs2=$(APXS)
+CONFIGURE_OPTIONS+=	--enable-diagnostics
+
+REQUIRED_PACKAGES += web/server/apache-24
+REQUIRED_PACKAGES += text/gnu-grep
```
Comment (Mno-hime) [inline components/web/apache2-modules/mod_auth_mellon/Makefile]: Wrt GNU grep in mellon_create_metadata.sh: If GNU grep is required to build the package, then it should be in `REQUIRED_PACKAGES`. If it's required only on runtime, one adds manual dependency like this https://github.com/OpenIndiana/oi-userland/blob/10f5892f9ab4d7cb51b52631b9f5b046b5400a97/components/archiver/zstd/zstd.p5m#L24

So, what is it?

**Comment by grueni** [comment]
Yesterday I was too tired. Of course you are right it is a runtime dependency.

**Review by Mno-hime — APPROVED** [review]

**Review by alarcher — APPROVED** [review]

**Pull request merged**